platform/settings.cpp

```diff
@@ -25,9 +25,11 @@ char const * kMeasurementUnits = "Units";
 
 StringStorage::StringStorage() : StringStorageBase(GetPlatform().SettingsPathForFile(SETTINGS_FILE_NAME)) {}
 
-StringStorage & StringStorage::Instance()
+StringStorage & StringStorage::Instance(bool reload)
 {
   static StringStorage inst;
+  if (reload)
```

Comment:
It's not good. StringStorage is thread-safe, but Reload() you doing here is not thread-safe. Moreover, existing invariants that assume that StringStorage can't be re-loaded may be broken. It's much better to investigate what's really happening and who calls Instance(), instead of this approach.